lib/validation.rb

```diff
@@ -304,4 +304,12 @@ def self.validate_from_option_tree(option_tree, option_parents, params)
       end
     end
   end
+
+  def self.validate_postgres_upgrade(version_int:, needs_convergence:, ongoing_failover:, read_replica:, needs_upgrade:)
```

Comment:
Have this accept a single `PostgresResource` argument, unless you have callers where you do not have a `PostgresResource` argument (I didn't see any).